delfin/drivers/dell_emc/vmax/client.py

```diff
@@ -68,12 +93,14 @@ def init_connection(self, access_info):
             LOG.error(msg)
             raise exception.StorageBackendException(msg)
 
-    def get_model(self):
+    def get_array(self):
         try:
-            # Get the VMAX model
+            # Get the VMAX array details (model, ucode )
             uri = "/system/symmetrix/" + self.array_id
-            model = self.conn.common.get_request(uri, "")
-            return model['symmetrix'][0]['model']
+            array_details = self.conn.common.get_request(uri, "")
+            array = {"model": array_details['symmetrix'][0]['model'],
+                     "ucode": array_details['symmetrix'][0]['ucode']}
+            return array
         except Exception as err:
             msg = "Failed to get model from VMAX: {}".format(err)
```

Comment:
This Failed message is only for model. what about if "ucode" got some error?

Reply:
Thanks, this PR is getting closed as the vmax driver re write has included array_id related chage. for ucode fetching I will raise another PR.